sensors/sensorlib/sensor_app.py

```diff
@@ -0,0 +1,177 @@
+from typing import Any
+from .signal_generator import (
+    WaveConfig,
+    NoiseConfig,
+)
+from .sensor_simualtor import (
+    SignalType,
+    Sensor,
+)
+from .rabbitmq_connector import (
+    MessageType,
+    RMQConfig,
+    SimpleRMQTopicConnection,
+)
+from flask import (
+    Flask,
+    abort,
+)
+from flask_script import Manager, Server
+from flask_restful import Api, Resource, reqparse, fields, marshal
+
+sensorConfigType = {
+    'sensorType': str,
+    'cycle': float,
+    'noiseMean': float,
+    'noiseStd': float,
+    'waveMagnitude': float,
+    'wavePoint': int
+}
+
+sensorConfigFields = {
+    'sensorType': fields.String,
+    'cycle': fields.Float,
+    'noiseMean': fields.Float,
+    'noiseStd': fields.Float,
+    'waveMagnitude': fields.Float,
+    'wavePoint': fields.Integer
+}
+
+DEFAULT_SENSOR_CONFIG: sensorConfigType = {
+    'sensorType': 'StaticWave',
+    'cycle': 10.0,
+    'noiseMean': 0.0,
+    'noiseStd': 0.3,
+    'waveMagnitude': 10.0,
+    'wavePoint': 20    
+}
+
+sensorConfig: sensorConfigType = DEFAULT_SENSOR_CONFIG
+
+class CustomServer(Server):
+    def __call__(self, app, *args, **kwargs):
+        sensorApp.startSensor()
+        sensorApp.sensorIsStart = True
+        return Server.__call__(self, app, *args, **kwargs)
+
+class SensorConfigAPI(Resource):
+    def __init__(self):
+        self.reqparse = reqparse.RequestParser()
+        self.reqparse.add_argument('sensorType', type=str, required=True, location='json')
+        self.reqparse.add_argument('cycle', type=float, required=True, location='json')
+        self.reqparse.add_argument('noiseMean', type=float, required=True, location='json')
+        self.reqparse.add_argument('noiseStd', type=float, required=True, location='json')
+        self.reqparse.add_argument('waveMagnitude', type=float, required=True, location='json')
+        self.reqparse.add_argument('wavePoint', type=int, required=True, location='json')
+        super(SensorConfigAPI, self).__init__()
+
+    def get(self):
+        return marshal(sensorApp.sensorConfig, sensorConfigFields)
+
+    def put(self):
+        args = self.reqparse.parse_args()
+        for key, val in args.items():
+            sensorConfig[key] = val
+
+        sensorApp.updateConfigSensor(sensorConfig)
+
+        return marshal(sensorConfig, sensorConfigFields)
+            
+
+class SensorApplication:
+    def __init__(self, sensorConfig: sensorConfigType):
+        self.__connector: SimpleRMQTopicConnection = SimpleRMQTopicConnection(RMQConfig(
+            "rabbitmq3",
```

Comment:
Don't hard code the parameters of RMQConfig. You should get them from json or ymal  file

Reply:
+1 move nó ra config nha fence